### DISCUSSION TITLE
Bootstrapping new cluster with existing repository and cyclical custom resource definitions

### DISCUSSION THREAD

**Discussion by luke-barnett** [fluxcd/flux2#4364] [Q&A] [answered]
I've been tearing down and building up k3s clusters to validate a build-out and have had trouble getting flux to bootstrap up a new cluster with a previously defined repository and the issue looks to be a somewhat cyclical failure that I'm interested in finding out if there is just a better pattern for.

A good example is cert-manager, which I have defined and installed as a HelmRepository and HelmRelease via flux. I then have a handful of "cert-manager.io/v1" certificate definitions based on the cert-manager CRD. However, it looks like the bootstrap command fails as the Kustomization doesn't reconcile as it can't create the certificate definitions as the CRD hasn't been applied. It then doesn't run any HelmReleases which means the CRD are never defined.

My understanding/expectation is that I should be able to take this repo and bootstrap a given cluster and it works itself out. Is there something I need to do differently for this to work in this fashion? Or do I just have a misunderstanding on what I should expect to be able to do with fluxcd?

**Comment by stefanprodan** [comment, marked as answer]
Please see the example on how to structure your repo so that Flux can install controllers, crds, etc in the right order: https://github.com/fluxcd/flux2-kustomize-helm-example

**Reply by luke-barnett** [reply]
perfect, thank you.
